litellm/llms/huggingface/chat/transformation.py

```diff
@@ -209,381 +97,67 @@ def transform_request(
-        raw_response: httpx.Response,
-        model_response: ModelResponse,
-        logging_obj: LoggingClass,
-        request_data: Dict,
-        messages: List[AllMessageValues],
-        optional_params: Dict,
-        litellm_params: Dict,
-        encoding: Any,
-        api_key: Optional[str] = None,
-        json_mode: Optional[bool] = None,
-    ) -> ModelResponse:
-        ## Some servers might return streaming responses even though stream was not set to true. (e.g. Baseten)
-        task = litellm_params.get("task", None)
-        is_streamed = False
-        if (
-            raw_response.__dict__["headers"].get("Content-Type", "")
-            == "text/event-stream"
-        ):
-            is_streamed = True
-
-        # iterate over the complete streamed response, and return the final answer
-        if is_streamed:
-            completion_response = self._convert_streamed_response_to_complete_response(
-                response=raw_response,
-                logging_obj=logging_obj,
-                model=model,
-                data=request_data,
-                api_key=api_key,
-            )
-        else:
-            ## LOGGING
-            logging_obj.post_call(
-                input=request_data,
-                api_key=api_key,
-                original_response=raw_response.text,
-                additional_args={"complete_input_dict": request_data},
-            )
-            ## RESPONSE OBJECT
-            try:
-                completion_response = raw_response.json()
-                if isinstance(completion_response, dict):
-                    completion_response = [completion_response]
-            except Exception:
-                raise HuggingfaceError(
-                    message=f"Original Response received: {raw_response.text}",
-                    status_code=raw_response.status_code,
-                )
 
-        if isinstance(completion_response, dict) and "error" in completion_response:
-            raise HuggingfaceError(
-                message=completion_response["error"],  # type: ignore
-                status_code=raw_response.status_code,
-            )
-        return self.convert_to_model_response_object(
-            completion_response=completion_response,
-            model_response=model_response,
-            task=task if task is not None and task in hf_task_list else None,
-            optional_params=optional_params,
-            encoding=encoding,
-            messages=messages,
-            model=model,
+    def transform_stream_chunk(self, chunk: Any) -> ModelResponse:
+        """Transform HF's ChatCompletionStreamOutput to ModelResponse"""
+        # Create streaming choices
+        streaming_choices = []
+        for choice in chunk.choices:
+            streaming_choices.append(dataclasses.asdict(choice))
+
+        # Create ModelResponse for streaming
+        response = ModelResponse(
+            id=chunk.id,
+            choices=streaming_choices,
```

Comment:
```suggestion
        # Create ModelResponse for streaming
        response = ModelResponse(
            id=chunk.id,
            choices=[dataclasses.asdict(choice) for choice in chunk.choices],
```

(nit) feels more readable to me